chef/cookbooks/provisioner/recipes/ssh.rb

```diff
@@ -0,0 +1,90 @@
+# Copyright 2011, Dell
+# Copyright 2015, SUSE Linux GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+directory "/root/.ssh" do
+  owner "root"
+  group "root"
+  mode "0700"
+  action :create
+end
+
+node.set["crowbar"]["ssh"] ||= {}
+
+# Start with a blank slate, to ensure that any keys removed from a
+# previously applied proposal will be removed.  It also means that any
+# keys manually added to authorized_keys will be automatically removed
+# by Chef.
+node.set["crowbar"]["ssh"]["access_keys"] = {}
+
+# Build my key
+if ::File.exists?("/root/.ssh/id_rsa.pub") == false
+  %x{ssh-keygen -t rsa -f /root/.ssh/id_rsa -N ""}
+end
+
+str = %x{cat /root/.ssh/id_rsa.pub}.chomp
```

Comment:
Style/UnneededPercentX: Do not use `%x` unless the command string contains backquotes.